tests/poll.rs

```diff
@@ -201,6 +205,98 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+#[test]
+fn registry_operations_are_thread_safe() {
+    init();
+
+    let mut poll = Poll::new().unwrap();
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let mut events = Events::with_capacity(128);
+
+    let mut listener1 = TcpListener::bind(any_local_address()).unwrap();
+    let addr = listener1.local_addr().unwrap();
+    let mut listener2 = TcpListener::bind(any_local_address()).unwrap();
+    let barrier = Arc::new(Barrier::new(4));
+
+    let registry1 = Arc::clone(&registry);
+    let registry2 = Arc::clone(&registry);
+    let registry3 = Arc::clone(&registry);
+    let barrier1 = Arc::clone(&barrier);
+    let barrier2 = Arc::clone(&barrier);
+    let barrier3 = Arc::clone(&barrier);
+
+    let handle1 = thread::spawn(move || {
+        registry1
+            .register(&mut listener1, ID1, Interest::READABLE)
+            .unwrap();
+        registry1.deregister(&mut listener1).unwrap();
+        registry1
+            .register(&mut listener1, ID1, Interest::READABLE)
+            .unwrap();
+        registry1
+            .reregister(
+                &mut listener1,
+                ID1,
+                Interest::READABLE.add(Interest::WRITABLE),
```

Comment:
`WRITABLE` makes no sense for a `TcpListener`.

Reply:
don't really need it, removed.